Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,28 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_identities:
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint16``,
+which means the maximum limit for a security identity is ``2^16 - 1``. The
+minimum security identity is ``256``, because reserve the identities from ``1``
+to ``255``.
+
+There are two caveats to the above however:
+
+1) ClusterMesh
+2) Identities generated from CIDR-based policies
+
+1) TODO: Explain ClusterMesh
```

Comment:
Clustermesh is the same, we just add 8 bits on top to identify the cluster. `cluster-id` is used for this.

So in effect we have:

0x00000000 - 0x0000FFFF => local identities
0x00010000 - 0x00FFFFFF => identities for remote clusters
0x01000000 - 0x0100FFFF => identities for CIDRs (node-local)

Reply:
Or, put another way, the bits at 0x00FF0000 represent the cluster-id of that identity.